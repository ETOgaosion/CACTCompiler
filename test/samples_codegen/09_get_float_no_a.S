iobuff:
	.zero 80
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	addi	sp, sp, -16
	call	get_float
	addi	sp, sp, 16
	fsw	fa0, -12(sp)
	flw	fa0, -12(sp)
	fld	fa0, -12(sp)
	addi	sp, sp, -16
	call	print_float
	addi	sp, sp, 16
	mv	a0, zero
	ld	ra, -8(sp)
	ret
