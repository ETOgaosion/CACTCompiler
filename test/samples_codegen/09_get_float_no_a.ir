============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
----------- tempVars ------------
t_0:= symbol type: TEMP VAR; data type: 3
----------- Codes ------------
main:
	call get_float; 
	t_0 =  return ;
	addparam t_0;
	call print_float; 
	return 0;

