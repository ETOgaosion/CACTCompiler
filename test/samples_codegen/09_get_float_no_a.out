bbl loader
please enter an float number:
4.540000
